From f09ec3d92fd9638e8c7c72eaf522ec1afb42037b Mon Sep 17 00:00:00 2001
From: Antonio Terceiro <terceiro@softwarelivre.org>
Date: Tue, 3 Feb 2015 10:35:17 -0200
Subject: [PATCH] linux/cap/halt: don't wait for `shutdown -h now` to finish

When running a Debian 8 lxc guest (with the vagrant-lxc plugin), which
has systemd as init system, `vagrant halt` will hang waiting for
`shutdown -h now` to return.

Signed-off-by: Johannes Kastl <kastl@b1-systems.de>
---
 plugins/guests/linux/cap/halt.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/guests/linux/cap/halt.rb b/plugins/guests/linux/cap/halt.rb
index 60dc5dde4..657636eaf 100644
--- a/plugins/guests/linux/cap/halt.rb
+++ b/plugins/guests/linux/cap/halt.rb
@@ -4,7 +4,7 @@ module VagrantPlugins
       class Halt
         def self.halt(machine)
           begin
-            machine.communicate.sudo("shutdown -h now")
+            machine.communicate.sudo("shutdown -h now &")
           rescue IOError, Vagrant::Errors::SSHDisconnected
             # Do nothing, because it probably means the machine shut down
             # and SSH connection was lost.
-- 
2.19.1

